commands/list.rb

```diff
@@ -0,0 +1,49 @@
+require 'fileutils'
+require_relative '../lib/color'
+require_relative '../lib/const'
+require_relative '../lib/package'
+
+class Command
+  def self.list(available, installed, compatible, incompatible, verbose)
```

Comment:
Use one argument?

Reply:
This saves us having to use an additional `if` statement in `bin/crew`:

One argument:
```ruby
# in bin/crew
def list_command(args)
  if args['available']
    Command.list(args['available'])
  elsif args['installed']
    Command.list(args['compatible'], @opt_verbose)
  elsif args['compatible']
    Command.list(args['compatible'])
  elsif args['incompatible']
    Command.list(args['incompatible'])
end
  
# in commands/list.rb
def self.list(args, verbose)
  when args
  case 'available'
    ...
  case 'installed'
    if verbose
      ...
    else
      ...
    end
  case 'compatible'
    ...
  case 'incompatible'
    ...
  end
end
```

Multiple arguments:
```ruby
# in bin/crew
def list_command(args)
  Command.list(args['available'], args['installed'], args['compatible'], args['incompatible'], @opt_verbose)
end
  
# in commands/list.rb
def self.list(available, installed, compatible, incompatible, verbose)
  if available
    ...
  elsif installed
    if verbose
      ...
    else
      ...
    end
  elsif compatible
    ...
  elsif incompatible
    ...
  end
end
```

Reply:
I don't see how that is less complicated.  Just pass `args` and then use the case statement.  No need to list out all the arguments as parameters.

Reply:
My end goal is that every `*_command(args)` function is a one-line call to the relevant command. 

I want to have the minimum possible amount of logic in `bin/crew`.

I can see no downside to having multiple parameters, but I do see the upside of saving lines of code and complexity that they allow.

Reply:
```
# in bin/crew
def list_command(args)
  Command.list(args)
end
  
# in commands/list.rb
def self.list(args)
  args.each do |arg|
    case arg
    when '-v', '--verbose'
      verbose = true
    end
  end
  args.each do |arg|
    case arg
    when 'available'
      ...
    when 'installed'
      if verbose
        ...
      else
        ...
      end
    when 'compatible'
      ...
    when 'incompatible'
      ...
    end
  end
end
```

Reply:
We use docopt for a reason-- moving argument handling logic to each command file and duplicating it does not seem like a good solution.

Reply:
I guess I'm not understanding the docopt process.  Seems counter intuitive.  Passing an entire list of parameters instead of a single parameter array is certainly  not a best practice.

Reply:
Parts of docopt are certainly not self-explanatory, this is the best solution I found that optimises for code simplicity and length. I agree the list of parameters is not ideal, but I'd prefer it over the other options.